noir-projects/noir-contracts/contracts/nft_contract/src/test/transfer_to_private.nr

```diff
@@ -0,0 +1,92 @@
+use crate::test::utils;
+use dep::aztec::{
+    hash::pedersen_hash, keys::getters::get_current_public_keys, prelude::{AztecAddress, NoteHeader},
+    oracle::unsafe_rand::unsafe_rand, protocol_types::storage::map::derive_storage_slot_in_map
+};
+use crate::{types::nft_note::NFTNote, NFT};
+
+#[test]
+unconstrained fn transfer_to_private_to_self() {
+    // The transfer to private to self is done in `utils::setup_mint_and_transfer_to_private` and for this reason
+    // in this test we just call it and check the outcome.
+
+    // Setup without account contracts. We are not using authwits here, so dummy accounts are enough
+    let (env, nft_contract_address, user, _, token_id) = utils::setup_mint_and_transfer_to_private(/* with_account_contracts */ false);
+
+    // User should have the note in their private nfts
+    utils::assert_owns_private_nft(nft_contract_address, user, token_id);
+
+    // Since the NFT was sent to private, the public owner should be zero address
+    utils::assert_owns_public_nft(env, nft_contract_address, AztecAddress::zero(), token_id);
+}
+
+#[test]
+unconstrained fn transfer_to_private_to_a_different_account() {
+    // Setup without account contracts. We are not using authwits here, so dummy accounts are enough
+    let (env, nft_contract_address, sender, recipient, token_id) = utils::setup_and_mint(/* with_account_contracts */ false);
+
+    let note_randomness = unsafe_rand();
+    let transient_storage_slot_randomness = unsafe_rand();
+    // Sender will be the msg_sender/transfer_preparer in prepare_transfer_to_private
+    let transfer_preparer_storage_slot_commitment = pedersen_hash(
+        [sender.to_field(), transient_storage_slot_randomness],
+        NFT::TRANSIENT_STORAGE_SLOT_PEDERSEN_INDEX
+    );
+
+    // We prepare the transfer
+    let prepare_transfer_to_private_call_interface = NFT::at(nft_contract_address).prepare_transfer_to_private(
+        sender,
+        recipient,
+        note_randomness,
+        transient_storage_slot_randomness
+    );
+    env.call_private_void(prepare_transfer_to_private_call_interface);
+
+    // Finalize the transfer of the NFT
+    let finalize_transfer_to_private_call_interface = NFT::at(nft_contract_address).finalize_transfer_to_private(token_id, transfer_preparer_storage_slot_commitment);
+    env.call_public(finalize_transfer_to_private_call_interface);
```

Comment:
This is quite hard to follow. Simply inlining the intermediate variable makes it more readable in my opinion:

```rust
env.call_public(
  NFT::at(nft_contract_address)
    .finalize_transfer_to_private(token_id, transfer_preparer_storage_slot_commitment)
);
```

Reply:
Agree, addressed where it made sense in https://github.com/AztecProtocol/aztec-packages/pull/8129/commits/3432498aaf2495e5f5e641f80e7644aa1b311685